component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordChangeHandler.java

```diff
@@ -178,4 +218,123 @@ public Map<String, String> getDefaultPropertyValues(String[] propertyNames, Stri
             throws IdentityGovernanceException {
         return null;
     }
+
+    /**
+     * Check whether the password is expired or not.
+     * @param tenantDomain user tenant domain
+     * @param tenantAwareUsername The tenant aware username of the user trying to authenticate
+     * @param userStoreManager
+     * @return true if password is expired or password last update time is null
+     * @throws org.wso2.carbon.user.api.UserStoreException
+     * @throws ParseException
+     */
+    private boolean isPasswordExpired(String tenantDomain, String tenantAwareUsername,
+                                      UserStoreManager userStoreManager) throws
+            org.wso2.carbon.user.api.UserStoreException {
+
+        String passwordLastChangedTime = getPasswordLastChangeTime(tenantDomain, tenantAwareUsername, userStoreManager);
+        int passwordExpiryInDays =  getPasswordExpiryInDaysConfig(tenantDomain);
+
+        long passwordChangedTime = 0;
+        if (passwordLastChangedTime != null) {
+            passwordChangedTime = Long.parseLong(passwordLastChangedTime);
+        }
+        int daysDifference = 0;
+        long currentTimeMillis = System.currentTimeMillis();
+        if (passwordChangedTime > 0) { // obtain the day difference from last password changed time to current time
+            Calendar currentTime = Calendar.getInstance();
+            currentTime.add(Calendar.DATE, (int) currentTime.getTimeInMillis());
+            daysDifference = (int) ((currentTimeMillis - passwordChangedTime) / (1000 * 60 * 60 * 24)); // convert to
```

Comment:
Not a security concern, but why are we computing `(1000 * 60 * 60 * 24)` for each request. Shouldn't this be a constant like, which also improves readability. 

why not use `TimeUnit.DAYS.toMillis(1)` to get the value for the constant, which also improves readability.